pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1080,52 +1073,31 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
 			CommonTlsContext: &auth.CommonTlsContext{},
 			Sni:              tls.Sni,
 		}
-		// Fallback to file mount secret instead of SDS
-		if !node.Metadata.SdsEnabled {
+
 			tlsContext.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{
 				ValidationContext: certValidationContext,
 			}
 			tlsContext.CommonTlsContext.TlsCertificates = []*auth.TlsCertificate{
 				{
 					CertificateChain: &core.DataSource{
 						Specifier: &core.DataSource_Filename{
-							Filename: model.GetOrDefault(proxy.Metadata.TLSClientCertChain, tls.ClientCertificate),
```

Comment:
This needs to be retained for people setting the TLSCertchain in ProxyMetadata

Reply:
So this is actually a bug. The code here is for user provided certs (via destination rules) and not for istio mTLS.